gui/tiletypes.lua

```diff
@@ -0,0 +1,1338 @@
+TileConfig.ATTRS {
+    on_change_shape=DEFAULT_NIL,
+    on_change_mat=DEFAULT_NIL,
+    on_change_stone=DEFAULT_NIL,
+    on_change_vein_type=DEFAULT_NIL,
+    on_change_special=DEFAULT_NIL,
+    on_change_variant=DEFAULT_NIL,
+}
+
+function TileConfig:init()
+    self.stone_enabled = false
+
+    for _,value in pairs(vein_type_list) do
+        value.label = {{ text= value.base_label, pen= value.base_pen } }
+    end
+
+    self:addviews {
+        widgets.CycleHotkeyLabel {
+            frame={l=1, t=0},
+            key_back='CUSTOM_SHIFT_H',
+            key='CUSTOM_H',
+            label='Shape:',
+            options=shape_list,
+            initial_option=-1,
+            on_change=self.on_change_shape,
+        },
+        widgets.Divider { frame={t=2}, frame_style_l=false, frame_style_r=false, },
+        widgets.CycleHotkeyLabel {
+            frame={l=1, t=4},
+            key_back='CUSTOM_SHIFT_J',
+            key='CUSTOM_J',
+            label='Material:',
+            options=mat_list,
+            initial_option=-1,
+            on_change=self.on_change_mat,
+        },
+        widgets.HotkeyLabel {
+            view_id="stone_label",
+            frame={l=3, t=6},
+            key='CUSTOM_N',
+            label={ text= "Stone:", pen=function() return self.stone_enabled and UI_COLORS.HIGHLIGHTED or UI_COLORS.DESELECTED end },
+            enabled=function() return self.stone_enabled end,
+            on_activate=function() self:openStonePopup() end
+        },
+        CycleLabel {
+            view_id="vein_type_label",
+            frame={l=3, t=8},
+            key='CUSTOM_I',
+            base_label={ text= "Vein Type:", pen=function() return self.stone_enabled and UI_COLORS.HIGHLIGHTED or UI_COLORS.DESELECTED end },
+            enabled=function() return self.stone_enabled end,
+            initial_option=-1,
+            options=vein_type_list,
+            on_change=self.on_change_vein_type,
+        },
+        widgets.Divider { frame={t=10}, frame_style_l=false, frame_style_r=false, },
+        widgets.CycleHotkeyLabel {
+            frame={l=1, t=12},
+            key_back='CUSTOM_SHIFT_K',
+            key='CUSTOM_K',
+            label='Special:',
+            options=special_list,
+            initial_option=-1,
+            on_change=self.on_change_special,
+        },
+        widgets.Divider { frame={t=14}, frame_style_l=false, frame_style_r=false, },
+        widgets.CycleHotkeyLabel {
+            frame={l=1, t=16},
+            key_back='CUSTOM_SHIFT_L',
+            key='CUSTOM_L',
+            label='Variant:',
+            options=variant_list,
+            initial_option=-1,
+            on_change=self.on_change_variant,
+        },
+        widgets.Divider { frame={t=18}, frame_style_l=false, frame_style_r=false, },
+    }
+
+    self:changeStone(-1)
+    self:setVisibility(self.visible)
+end
+
+function TileConfig:openStonePopup()
+    local dialog = guimat.MaterialDialog {
```

Comment:
As strange as that might be, those are actually all valid materials. Try it out with the command line `tiletypes` and you'll notice they're accepted and handled correctly, since they're valid inorganic stone materials, even if they seem like they shouldn't be.
Divine metals, evil clouds, and evil rains can all be generated with the necessary attributes to be qualified, but not all of them necessarily will be. Only the qualified ones should show up in that list though.

Reply:
A couple examples from my testing map:
![image](https://github.com/DFHack/scripts/assets/26845479/6b5c628f-8887-4897-a59f-a275b35595f6)

Reply:
True, they're *possible* to use, but for the common case, they'll just get in the way. Maybe we can do something similar to `gui/create-item`, where the "expected" materials are shown by default, but you can pass `--unrestricted` when launching the UI to see all materials.